pkg/controllers/dynakube/connectioninfo/oneagent/reconciler.go

```diff
@@ -62,36 +71,35 @@ func (r *reconciler) Reconcile(ctx context.Context) error {
 
 func (r *reconciler) reconcileConnectionInfo(ctx context.Context) error {
 	secretNamespacedName := types.NamespacedName{Name: r.dynakube.OneagentTenantSecret(), Namespace: r.dynakube.Namespace}
-	isOutdated := r.dynakube.IsOneAgentConnectionInfoUpdateAllowed(r.timeProvider)
 
-	if !isOutdated {
+	if !conditions.IsOutdated(r.timeProvider, r.dynakube, oaConnectionInfoConditionType) {
 		isSecretPresent, err := connectioninfo.IsTenantSecretPresent(ctx, r.apiReader, secretNamespacedName, log)
 		if err != nil {
 			return err
 		}
 
+		condition := meta.FindStatusCondition(*r.dynakube.Conditions(), oaConnectionInfoConditionType)
 		if isSecretPresent {
 			log.Info(dynatracev1beta1.GetCacheValidMessage(
 				"OneAgent connection info update",
-				r.dynakube.Status.OneAgent.ConnectionInfoStatus.LastRequest,
+				condition.LastTransitionTime,
 				r.dynakube.FeatureApiRequestThreshold()))
 
 			return nil
 		}
 	}
 
+	conditions.SetSecretOutdatedCondition(r.dynakube.Conditions(), oaConnectionInfoConditionType, secretNamespacedName.Name+" is not present or outdated, update in progress") // Necessary to update the LastTransitionTime, also it is a nice failsafe
```

Comment:
```suggestion
	conditions.SetOutdatedSecretCondition(r.dynakube.Conditions(), oaConnectionInfoConditionType, secretNamespacedName.Name+" is not present or outdated, update in progress") // Necessary to update the LastTransitionTime, also it is a nice failsafe
```
just easy to read :)

Reply:
should we switch the word order for the for the other `Secret` related condtions?
like `CreatedSecret`, `UpdatedSecret`, `OutdatedSecret`

I went with `SecretOutdated` for only consistency 😅

Reply:
oh....